### PR TITLE
Add IMDSv2 support to FAQ

### PR DESCRIPTION
**Pull request by torredil**
**Is this a bug fix or adding new feature?**

Documentation

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -172,6 +172,12 @@ Containers:
       --timeout=61s
 ```
 
+## IMDSv2 Support
+
+The driver supports the use of [IMDSv2](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) (Instance Metadata Service Version 2) for enhanced security when accessing EC2 instance metadata.
```
Comment (ConnorJC3) [inline docs/faq.md:177]: Drop the words "for enhanced security when accessing EC2 instance metadata", as our next paragraph has you completely eliminate most/all those security benefits.

File: docs/faq.md [hunk 2]
```diff
@@ -172,6 +172,12 @@ Containers:
       --timeout=61s
 ```
 
+## IMDSv2 Support
```
Comment (ConnorJC3) [inline docs/faq.md:175]: I think we should strongly consider including a line in this section suggesting the use of Kubernetes metadata when possible - IMDS metadata requires configuring IMDS in an insecure manner (by increasing the hop limit).

File: docs/faq.md (hunk 1 above)
Comment (torredil) [reply docs/faq.md:177]: Dropped!

File: docs/faq.md (hunk 2 above)
Comment (torredil) [reply docs/faq.md:175]: Hesitant to provide that recommendation given Kubernetes metadata does not have information about attached ENIs ([NumAttachedENIs](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/85563fbdcb31e7234abf9f9a335008ee0c70c5b8/pkg/cloud/metadata_k8s.go#L82) is hard-coded to 1) and thus the allocatable count as reported by `NodeGetInfo` may be incorrect. That said, can add this recommendation in next revision if you strongly recommend it.

Comment (ConnorJC3) [reply docs/faq.md:175]: Fair enough

**Review by AndrewSirenko — APPROVED** [review]
This is great-- prevents future users from running into this papercut, thank you!

**Pull request merged**